### PR TITLE
Update Pydantic Version 

### PR DESCRIPTION
**Pull request by THOR300**
### This Pull Request: 
---
- Integrates an update to the pydantic version in the data access lib from 1 -> 2. 
- This work was triggered as we had to have to implement a sub optimal solution in another PR [here](https://github.com/climatepolicyradar/data-access/pull/87)
- It is also an update that just has to be done at some point. 
- Please see the [Migration Guide](https://docs.pydantic.dev/latest/migration/#changes-to-pydanticbasemodel).

_The discovery work on this was done in this PR [HERE](https://github.com/climatepolicyradar/data-science-research/pull/51)_

Also: 
- Integrates a new test that shows the correct behaviour of the exception handling
- Removes two deprecation warnings and related functions for the `to_json` methods. 
- Some slight code changes in the `models` directory that should be closely inspected.

### CODE REVIEW

File: src/cpr_data_access/models/__init__.py [hunk 1]
```diff
@@ -920,13 +924,25 @@ def add_metadata(
                         for s in new_metadata_dict.get("Sectors", "").split(";")
                     ],
                     status=new_metadata_dict.pop("CPR Document Status"),
-                    collection_id=new_metadata_dict.pop("CPR Collection ID"),
-                    collection_name=new_metadata_dict.pop("Collection name"),
+                    collection_id=(
```
Comment (THOR300) [inline src/cpr_data_access/models/__init__.py:927]: Area of note @kdutia

File: src/cpr_data_access/models/__init__.py [hunk 2]
```diff
@@ -956,12 +972,20 @@ def add_metadata(
                     date=new_metadata_dict.pop("Date"),
                     link=new_metadata_dict.pop("Documents"),
                     author_is_party=new_metadata_dict.pop("Author Type") == "Party",
-                    collection_id=new_metadata_dict.pop("CPR Collection ID"),
+                    collection_id=(
```
Comment (THOR300) [inline src/cpr_data_access/models/__init__.py:975]: Area of note @kdutia

File: src/cpr_data_access/models/__init__.py [hunk 3]
```diff
@@ -1157,10 +1183,19 @@ def to_huggingface(
                 citation=citation or "",
             )
 
+        mapping = {
```
Comment (THOR300) [inline src/cpr_data_access/models/__init__.py:1186]: Area of note @kdutia

Comment (THOR300) [reply src/cpr_data_access/models/__init__.py:1186]: @harrisonpim

File: src/cpr_data_access/models/__init__.py (hunk 1 above)
Comment (THOR300) [reply src/cpr_data_access/models/__init__.py:927]: @harrisonpim

File: src/cpr_data_access/models/__init__.py (hunk 3 above)
Comment (harrisonpim) [reply src/cpr_data_access/models/__init__.py:1186]: what's going on with these `plain_urls`?

**Review by harrisonpim — APPROVED** [review]
looks good!

File: src/cpr_data_access/models/__init__.py (hunk 3 above)
Comment (THOR300) [reply src/cpr_data_access/models/__init__.py:1186]: The mapping from a Url object seemed to fail, not entirely sure why the pydantic update would have caused that!

**Pull request merged**